src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -510,71 +557,87 @@ private fun nodeIntroducesNewScope(
+                            receiverType.elementType
+                        } else if (isSwizzle(expression.memberName)) {
+                            Type.Vector(expression.memberName.length, receiverType.elementType)
+                        } else {
+                            TODO()
+                        }
+
+                    else -> throw UnsupportedOperationException("Member lookup not implemented for receiver of type $receiverType")
+                }
+
             when (val receiverType = resolverState.resolvedEnvironment.typeOf(expression.receiver)) {
-                is Type.Struct ->
-                    receiverType.members
-                        .firstOrNull {
-                            it.first == expression.memberName
-                        }?. second
-                        ?: throw IllegalArgumentException("Struct with type $receiverType does not have a member ${expression.memberName}")
-                is Type.Vector ->
-                    // In the following we could check whether the vector indices exist, e.g. using z on a vec2 is not be allowed.
-                    if (expression.memberName in setOf("x", "y", "z", "w", "r", "g", "b", "a")) {
-                        receiverType.elementType
-                    } else if (isSwizzle(expression.memberName)) {
-                        Type.Vector(expression.memberName.length, receiverType.elementType)
-                    } else {
-                        TODO()
-                    }
-                is Type.Pointer ->
-                    when (val pointeeType = receiverType.pointeeType) {
-                        is Type.Vector ->
-                            // In the following we could check whether the vector indices exist, e.g. using z on a vec2 is not be allowed.
-                            if (expression.memberName in setOf("x", "y", "z", "w")) {
-                                pointeeType.elementType
-                            } else if (isSwizzle(expression.memberName)) {
-                                Type.Vector(expression.memberName.length, pointeeType.elementType)
-                            } else {
-                                TODO()
-                            }
-                        is Type.Struct ->
-                            pointeeType.members
-                                .firstOrNull {
-                                    it.first == expression.memberName
-                                }?. second
-                                ?: throw IllegalArgumentException(
-                                    "Struct with type $receiverType does not have a member ${expression.memberName}",
-                                )
-                        else -> TODO("${receiverType.pointeeType}")
-                    }
-                else -> throw UnsupportedOperationException("Member lookup not implemented for receiver of type $receiverType")
+                is Type.Pointer -> {
```

Comment:
This looks good and broadly similar to the index lookup cases. I asked for quick examples there, so quick examples here would be good to, or a note that this is rather similar to the index lookup cases.

As per my question about whether one can have a pointer / reference to a matrix column, it looks here like a member lookup gives a pointer / reference to an element of a structure, right?